File: src/main/java/fr/univ_smb/isc/m1/brutocollinus/infrastructure/persistence/entity/BrutoClass.java
package fr.univ_smb.isc.m1.brutocollinus.infrastructure.persistence.entity;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToOne;

@Entity
public class BrutoClass extends Identifiable {
    private String name;

    @OneToOne(cascade = CascadeType.ALL)
    private FightStatistics fightStatistics;

    public BrutoClass() {
        // JPA
    }

    public BrutoClass(String name, FightStatistics fightStatistics) {
        this.name = name;
        this.fightStatistics = fightStatistics;
    }

    public String name() {
        return name;
    }

    public FightStatistics fightStatistics() {
        return this.fightStatistics;
    }
}

File: frontend/src/components/BrutoClass.js
import React from 'react'
import FightStatistics from './FightStatistics.js';

export default function BrutoClass(props) {
    const brutoClass = props.brutoClass;

    return (
        <div>
            {brutoClass.name}
            <FightStatistics fightStatistics={brutoClass.fightStatistics}/>
        </div>
    );
}